xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/Wrappers.java

```diff
@@ -283,6 +283,10 @@ public boolean equals(Object obj) {
      * @return {@link TransferFee}
      */
     static TransferFee ofPercent(BigDecimal percent) {
+      Preconditions.checkArgument(
+        Math.max(0, percent.stripTrailingZeros().scale()) <= 2,
+        "Percent value should have a maximum of 2 decimal places."
+        );
```

Comment:
This is fine and actually probably better, but we don't technically need this. `toBigIntegerExact()` will throw an `ArithmeticException` if you have too many decimal places. An `IllegalArgumentException` with our own message is probably better for developers to see though.